### PR TITLE
Add yaml file for CI with GitHub Actions

### PR DESCRIPTION
**Pull request by joshflchan**
- Inspired by [this comment](https://github.com/joshflchan/software-product-sprint/pull/4#issuecomment-639596519).
- Never worked with GitHub Actions, saw this as a fun learning opportunity.
- Followed the [GCP GitHub Actions repo](https://github.com/GoogleCloudPlatform/github-actions/tree/master/example-workflows/gae).

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+      - name: Setup gcloud CLI
+        uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+        with:
+          version: '290.0.1'
+          project_id: jchan-sps-summer20
+          service_account_key: ${{ secrets.GCLOUD_AUTH }}
+          
+      - name: Deploy with Maven
+        run: mvn package appengine:deploy -q
```
Comment (KevinWorkman) [inline .github/workflows/maven.yml:29]: Ohh does this all run locally when you push to master? That's clever.

**Comment by joshflchan** [comment]
@KevinWorkman Hahaha hopefully it works as anticipated when I hit merge! It worked when I pushed directly to this branch. I have it set to trigger on push to master, which I THINK also happens under the hood when a PR is merged. 

I could maybe write a blog post on the site and maybe share the link!

File: .github/workflows/maven.yml (hunk 1 above)
Comment (joshflchan) [reply .github/workflows/maven.yml:29]: From my understanding of GitHub Actions so far, I think what's happening is that the pipeline is run on a GitHub-hosted VM. I tried my best to mimic the environment in Cloud Shell, which is running Debian 9.2 I believe, but GitHub doesn't offer that, so I went with Ubuntu. 

The pipeline can then be broken down into **actions**. There are some common ones by GitHub such as checking out and setting up Java 8, which is what I used. [GCP also has a library for actions](https://github.com/GoogleCloudPlatform/github-actions), one of them being to set up a gcloud Cloud SDK environment and adding the CLI tool in the worker.

I had to enable the App Engine Admin API and a few other things (such as a secret key) for a service account within my SPS App Engine project. _This is where I'm a little unsure about the process_, but I think when the worker runs `mvn package appengine:deploy -q` it communicates with App Engine through the API to deploy the site via the worker.

I also found that there is an [action to deploy via gcloud](https://github.com/GoogleCloudPlatform/github-actions/tree/master/appengine-deploy), but ran into some issues previously about `Maven source deployment is not supported for Java8 GAE project`. Maybe I'll give that a try again some time later ahaha

**Comment by joshflchan** [comment]
> @KevinWorkman Hahaha hopefully it works as anticipated when I hit merge! It worked when I pushed directly to this branch. I have it set to trigger on push to master, which I THINK also happens under the hood when a PR is merged.
> 
> I could maybe write a blog post on the site and then share the link!

image

Looks like it was triggered!

**Pull request merged**